src/devices/sunricher.ts

```diff
@@ -361,7 +362,159 @@ async function syncTime(endpoint: Zh.Endpoint) {
+        e.enum('motion_sensor_microwave_switch', ea.SET, ['on', 'off']).withDescription('Motion sensor microwave switch (default: On)'),
+        e.enum('motion_sensor_onoff_broadcast', ea.SET, ['on', 'off']).withDescription('Motion sensor on/off broadcast (default: On)'),
+        e.enum('motion_sensor_light_state', ea.SET, ['on', 'off']).withDescription('Motion sensor light state (default: On)'),
+        e
+            .numeric('motion_sensor_in_pwm_brightness', ea.SET)
+            .withValueMin(0)
+            .withValueMax(1000)
+            .withUnit('lux')
+            .withDescription('Motion sensor IN PWM brightness (0-1000 lux, default: 0)'),
+        e
+            .numeric('motion_sensor_in_pwm_output', ea.SET)
+            .withValueMin(0)
+            .withValueMax(254)
+            .withDescription('Motion sensor IN PWM output (0-254, default: 254)'),
+        e
+            .numeric('motion_sensor_leave_pwm_output', ea.SET)
+            .withValueMin(0)
+            .withValueMax(100)
+            .withUnit('%')
+            .withDescription('Motion sensor LEAVE PWM output (0%-100%, default: 0%)'),
+        e
+            .numeric('motion_sensor_leave_delay', ea.SET)
+            .withValueMin(0)
+            .withValueMax(65535)
+            .withUnit('s')
+            .withDescription('Motion sensor LEAVE delay (0s-65535s, default: 0s)'),
+        e
+            .numeric('motion_sensor_pwm_output_after_delay', ea.SET)
+            .withValueMin(0)
+            .withValueMax(100)
+            .withUnit('%')
+            .withDescription('Motion sensor PWM output after delay (0%-100%, default: 0%)'),
+        e
+            .numeric('linear_error_ratio_coefficient_of_lux_measurement', ea.SET)
+            .withValueMin(100)
+            .withValueMax(10000)
+            .withDescription('Linear error ratio coefficient of LUX measurement (100‰-10000‰, default: 1000‰)'),
+        e
+            .numeric('fixed_deviation_of_lux_measurement', ea.SET)
+            .withValueMin(-32768)
+            .withValueMax(32767)
+            .withDescription('Fixed deviation of LUX measurement (-32768~32767, default: 0)'),
+    ];
+
+    return {
+        toZigbee,
+        exposes,
+        isModernExtend: true,
+    };
+}
+
 const definitions: DefinitionWithExtend[] = [
+    {
+        zigbeeModel: ['ZG9030A-MW'],
+        model: 'SR-ZG9030A-MW',
+        vendor: 'Sunricher',
+        description: 'Zigbee compatible ceiling mount occupancy sensor',
+        extend: [
+            sunricherZG9030AConfig(),
+            deviceEndpoints({endpoints: {'1': 1, '2': 2, '3': 3}}),
+            light(),
+            occupancy(),
+            illuminance({endpointNames: ['3']}),
+            commandsOnOff(),
+            commandsLevelCtrl(),
+        ],
+        meta: {multiEndpoint: true},
+        toZigbee: [sunricher.tz.setModel],
+        exposes: [e.enum('model', ea.SET, ['HK-DIM', 'ZG9030A-MW']).withDescription('Model of the device')],
```

Comment:
Where is this used for? (what does changing the `modelId` do)

Reply:
Some `ZG9030A-MW` devices were mistakenly set with the modelId `HK-DIM` during manufacturing. Therefore, we need to provide a method that allows users to update the modelId of these `HK-DIM` devices to `ZG9030A-MW`.